src/util/copyable.h

```diff
@@ -0,0 +1,47 @@
+#ifndef SRC_UTIL_COPYABLE_H_
+#define SRC_UTIL_COPYABLE_H_
+
+#include <utility>
+
+namespace raksha::util {
+
+template<class T>
+class Copyable {
```

Comment:
What is the advantage of having this class? 

If you want to mark something as not copyable, you can simply delete the copy constructor with `= delete`? See https://en.cppreference.com/w/cpp/language/function#Deleted_functions

Reply:
The idea of this class is to make things that I have made not implicitly copyable through making the copy constructor explicit and/or private silently copyable while wrapped in this class. Making the copy functions private while having public move functions in classes has allowed the compiler to catch unintentional copies for me.

I'm currently using this because I want to make some canonical values to use as parameters to my unit tests and place them in a `const` array. Some of those values are classes for which I have made the copy constructor private. This makes those values ineligible to be moved. However, because they are tests, we don't care very much that they are making copies and so this provides a way to get those tests to work.

Reply:
@bgogul convinced me that this was undermining the compiler's ability to handle copying for us. Restored the default copy constructors.